### PR TITLE
Default product template in categories

### PR DESCRIPTION
**Pull request by roadster31**
This PR add a  new feature: a default product template could be defined in categories. 

Products created in this category will get this default product template. If no default product template is defined in a given category, it will be searched in parent categories.

### CODE REVIEW

File: core/lib/Thelia/Controller/Admin/CategoryController.php [hunk 1]
```diff
@@ -24,6 +24,7 @@
 use Thelia\Core\Security\AccessManager;
 use Thelia\Core\Security\Resource\AdminResources;
 use Thelia\Form\Definition\AdminForm;
+use Thelia\Model\Category;
```
Comment (lunika) [inline core/lib/Thelia/Controller/Admin/CategoryController.php]: This class is never used

Comment (roadster31) [inline core/lib/Thelia/Controller/Admin/CategoryController.php]: It is. Line 119 (same old debate, isn't it ?)

File: tests/phpunit/Thelia/Tests/Action/CategoryTest.php [hunk 2]
```diff
@@ -16,9 +16,12 @@
 use Thelia\Core\Event\Category\CategoryDeleteEvent;
 use Thelia\Core\Event\Category\CategoryToggleVisibilityEvent;
 use Thelia\Core\Event\Category\CategoryUpdateEvent;
+use Thelia\Core\Event\Template\TemplateDeleteEvent;
 use Thelia\Model\Category as CategoryModel;
 use Thelia\Core\Event\Category\CategoryCreateEvent;
 use Thelia\Model\CategoryQuery;
+use Thelia\Model\Template;
+use Thelia\Model\TemplateQuery;
```
Comment (lunika) [inline tests/phpunit/Thelia/Tests/Action/CategoryTest.php]: this class is never used

**Comment by lunika** [comment]
please update the file ```setup/update/sql/2.2.0-alpha.sql``` with the new modification.

Thanks

**Pull request merged**